src/engine-actions/translate-swap.spec.js

```diff
@@ -57,47 +64,436 @@ describe('translate-swap', () => {
 
     afterEach(() => {
       timekeeper.reset()
+      reverts.forEach(revert => revert())
     })
 
-    it('sends a payment using the maximum time lock', async () => {
-      await translateSwap.call(engine, address, swapHash, amount, maxTime)
+    context('existing payment', () => {
+      it('gets the preimage if it exists', async () => {
+        const result = await translateSwap.call(engine, address, swapHash, amount, maxTime)
 
-      expect(sendPayment).to.have.been.calledOnce()
-      expect(sendPayment).to.have.been.calledWith(
-        {
-          paymentHash: swapHash,
-          destString: pubKey,
-          amt: amount,
-          finalCltvDelta: 144,
-          cltvLimit: 288,
-          feeLimit: 0
-        },
-        { client })
+        expect(getPreimageIfPaymentExists).to.have.been.calledOnce()
+        expect(getPreimageIfPaymentExists).to.have.been.calledWith(swapHash, engine.client, engine.logger)
+        expect(result).to.be.eql(preimage)
+      })
+    })
+
+    context('no existing payment', () => {
+      beforeEach(() => {
+        const error = new Error()
+        error.code = 5 // gRPC NOT FOUND status code
+        getPreimageIfPaymentExists.rejects(error)
+      })
+
+      it('throws an error for any error other than NOT FOUND', () => {
+        getPreimageIfPaymentExists.rejects()
+
+        return expect(translateSwap.call(engine, address, swapHash, amount, maxTime)).to.eventually.be.rejectedWith(Error)
```

Comment:
Might be better to test the error message itself since almost anything we throw will be an `Error`

Reply:
Might also be nice to test the pass through of Error types (`PermanentSwapError` and otherwise)